torchgen/gen_functionalization_type.py

```diff
@@ -656,6 +656,7 @@ def emit_inplace_functionalization_body(
         {return_type} tmp_output;
         {{
           at::AutoDispatchSkipFunctionalize guard;
+          c10::impl::IncludeDispatchKeyGuard include_guard(DispatchKey::PreDispatch);
```

Comment:
If we're going to have the `Functionalize` key (potentially) jump back to the `PreDispatch` key, it feels a bit better if we first check if the pre-dispatch mode stack has any modes on it first, and only add this guard if there are any pre-dispatch modes left

Reply:
Yep agreed. I was too lazy to figure out how to import TorchDispatchTLS here lol.

Reply:
done